spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/SCBackendService.java

```diff
@@ -548,107 +548,120 @@ private String buildDeviceInfo(String token) {
     }
 
     private void listenForSyncEvents() {
-
         Observable<SCDataStore> syncableStores = dataService.getISyncableStores();
 
-        Observable<Boolean>  storeEditSync = syncableStores
-                .flatMap(new Func1<SCDataStore, Observable<SCSpatialFeature>>() {
+        Observable<Object> storeEditSync =
+                syncableStores.flatMap(new Func1<SCDataStore, Observable<SCSpatialFeature>>() {
                     @Override
                     public Observable<SCSpatialFeature> call(SCDataStore scDataStore) {
                         return scDataStore.storeEdited;
                     }
-                }).map(new Func1<SCSpatialFeature, Boolean>() {
+                }).flatMap(new Func1<SCSpatialFeature, Observable<?>>() {
                     @Override
-                    public Boolean call(SCSpatialFeature feature) {
-                        final ISyncableStore store = (ISyncableStore) dataService.getStoreByIdentifier(feature.getStoreId());
-                        syncStore(store);
-                        return null;
+                    public Observable call(SCSpatialFeature feature) {
+                        final ISyncableStore store = (ISyncableStore) dataService
+                                .getStoreByIdentifier(feature.getStoreId());
+                        return syncStore(store);
                     }
                 });
 
         //sync all stores when connection to broker is true
-        Observable<Boolean> onlineSync = connectedToBroker
-                .filter(new Func1<Boolean, Boolean>() {
-                    @Override
-                    public Boolean call(Boolean sync) {
-                        return sync;
-                    }
-                }).map(new Func1<Boolean, Boolean>() {
-                    @Override
-                    public Boolean call(Boolean sync) {
-                        syncStores();
-                        return sync;
-                    }
-                });
+        Observable<Object> onlineSync = connectedToBroker.filter(new Func1<Boolean, Boolean>() {
+            @Override
+            public Boolean call(Boolean sync) {
+                return sync;
+            }
+            // for each connected State true (sync is true), sync the stores
+        }).flatMap(new Func1<Boolean, Observable<?>>() {
+            @Override
+            public Observable call(Boolean sync) {
+                return syncStores();
+            }
+        });
 
-        Observable<Boolean> sync = onlineSync.mergeWith(storeEditSync);
-        sync.subscribeOn(Schedulers.newThread()).subscribe();
-    }
+        Observable<Object> sync = onlineSync.mergeWith(storeEditSync);
+        sync.subscribeOn(Schedulers.newThread()).subscribe(new Action1<Object>() {
```

Comment:
This should use `Schedulers.io()` instead of a new thread. It will use a thread pool of already created threads instead of creating a new one.